# CIA World Factbook Project

The CIA Factbook contains various statistics about all of the countries in the world, including:

 - population — the global population.
 - population_growth — the annual population growth rate, as a percentage.
 - area — the total land and water area.

In [1]:
!pip install ipython-sql

You should consider upgrading via the 'c:\users\florence\anaconda3\python.exe -m pip install --upgrade pip' command.


In [2]:
%%capture
%load_ext sql
%sql sqlite:///factbook.db

In [3]:
%%sql
SELECT *
  FROM sqlite_master
 WHERE type='table';

 * sqlite:///factbook.db
Done.


[('table', 'sqlite_sequence', 'sqlite_sequence', 3, 'CREATE TABLE sqlite_sequence(name,seq)'),
 ('table', 'facts', 'facts', 47, 'CREATE TABLE "facts" ("id" INTEGER PRIMARY KEY AUTOINCREMENT NOT NULL, "code" varchar(255) NOT NULL, "name" varchar(255) NOT NULL, "area" integer, "a ... (4 characters truncated) ... land" integer, "area_water" integer, "population" integer, "population_growth" float, "birth_rate" float, "death_rate" float, "migration_rate" float)')]

In [4]:
%%sql
SELECT *
  FROM facts
 LIMIT 5;

 * sqlite:///factbook.db
Done.


[(1, 'af', 'Afghanistan', 652230, 652230, 0, 32564342, 2.32, 38.57, 13.89, 1.51),
 (2, 'al', 'Albania', 28748, 27398, 1350, 3029278, 0.3, 12.92, 6.58, 3.3),
 (3, 'ag', 'Algeria', 2381741, 2381741, 0, 39542166, 1.84, 23.67, 4.31, 0.92),
 (4, 'an', 'Andorra', 468, 468, 0, 85580, 0.12, 8.13, 6.96, 0.0),
 (5, 'ao', 'Angola', 1246700, 1246700, 0, 19625353, 2.78, 38.78, 11.49, 0.46)]

The table contains 8 columns; these 8 columns are:

 - name - name of the country
 - area - total land and water area of the country
 - area_land - land area of the country in square kilometres
 - area_water - water area of the country in square kilometres
 - population - population of the country
 - population_growth - how quickly the population of the country grows as a      percentage
 - birth_rate - number of births per 1,000 people 
 - death_rate - number of deaths per 1,000 people
 - migration_rate

In [5]:
%%sql
SELECT MIN(population),
       MAX(population),
       MIN(population_growth),
       MAX(population_growth)
  FROM facts;

 * sqlite:///factbook.db
Done.


[(0, 7256490011, 0.0, 4.02)]

The minimum population is zero, and the maximum population is approximately 7.2 billion, which is around the world's total population. The maximum popluation growth is 4.02%.

In [6]:
%%sql
SELECT *
  FROM facts
 WHERE population == (SELECT MIN(population)
                        FROM facts);

 * sqlite:///factbook.db
Done.


[(250, 'ay', 'Antarctica', None, 280000, None, 0, None, None, None, None)]

In [7]:
%%sql
SELECT *
  FROM facts
 WHERE population == (SELECT MAX(population)
                        FROM facts);

 * sqlite:///factbook.db
Done.


[(261, 'xx', 'World', None, None, None, 7256490011, 1.08, 18.6, 7.8, None)]

We need to exclude Antartica and the world from our analysis of the minimum and maximum population, as Antartica only has research scientists there, many of whom do not live there permanently, and the world row is for the entire world, which includes all of the countries on the list.

In [8]:
%%sql
SELECT MIN(population),
       MAX(population),
       MIN(population_growth),
       MAX(population_growth)
  FROM facts
 WHERE name <> 'World';

 * sqlite:///factbook.db
Done.


[(0, 1367485388, 0.0, 4.02)]

In [9]:
%%sql
SELECT AVG(population) AS avg_pop, AVG(area) AS avg_area
  FROM facts
 WHERE name <> 'World';

 * sqlite:///factbook.db
Done.


[(32242666.56846473, 555093.546184739)]

In [10]:
%%sql
SELECT *
  FROM facts
 WHERE population > (SELECT AVG(population)
                       FROM facts
                      WHERE name <>'World' AND name <> 'Antartica')
 AND area <  (SELECT AVG(area)
                FROM facts
               WHERE name <> 'World' AND name <> 'Antartica');

 * sqlite:///factbook.db
Done.


[(14, 'bg', 'Bangladesh', 148460, 130170, 18290, 168957745, 1.6, 21.14, 5.61, 0.46),
 (65, 'gm', 'Germany', 357022, 348672, 8350, 80854408, 0.17, 8.47, 11.42, 1.24),
 (80, 'iz', 'Iraq', 438317, 437367, 950, 37056169, 2.93, 31.45, 3.77, 1.62),
 (83, 'it', 'Italy', 301340, 294140, 7200, 61855120, 0.27, 8.74, 10.19, 4.1),
 (85, 'ja', 'Japan', 377915, 364485, 13430, 126919659, 0.16, 7.93, 9.51, 0.0),
 (91, 'ks', 'Korea, South', 99720, 96920, 2800, 49115196, 0.14, 8.19, 6.75, 0.0),
 (120, 'mo', 'Morocco', 446550, 446300, 250, 33322699, 1.0, 18.2, 4.81, 3.36),
 (138, 'rp', 'Philippines', 300000, 298170, 1830, 100998376, 1.61, 24.27, 6.11, 2.09),
 (139, 'pl', 'Poland', 312685, 304255, 8430, 38562189, 0.09, 9.74, 10.19, 0.46),
 (163, 'sp', 'Spain', 505370, 498980, 6390, 48146134, 0.89, 9.64, 9.04, 8.31),
 (173, 'th', 'Thailand', 513120, 510890, 2230, 67976405, 0.34, 11.19, 7.8, 0.0),
 (182, 'ug', 'Uganda', 241038, 197100, 43938, 37101745, 3.24, 43.79, 10.69, 0.74),
 (185, 'uk', 'United Kingdom', 243610, 241930, 1680, 64088222, 0.54, 12.17, 9.35, 2.54),
 (192, 'vm', 'Vietnam', 331210, 310070, 21140, 94348835, 0.97, 15.96, 5.93, 0.3)]

Some of these countries are known for being densely populated, however I'm surprised that neither China nor India is on this list, as they're both known for their large populations and being densely populated.

In [11]:
%%sql

SELECT name, population
  FROM facts
 WHERE name <> 'World' AND population == (SELECT MAX(population)
                                            FROM facts
                                            WHERE name <> 'World');

 * sqlite:///factbook.db
Done.


[('China', 1367485388)]

In [12]:
%%sql

SELECT name, population_growth
  FROM facts
 WHERE name <> 'World' AND population_growth == (SELECT MAX(population_growth)
                                            FROM facts
                                            WHERE name <> 'World');

 * sqlite:///factbook.db
Done.


[('South Sudan', 4.02)]

## Countries with highest water to land ratio

In [13]:
%%sql

SELECT name, CAST(area_water AS FLOAT) / area_land AS water_land_ratio
  FROM facts
 WHERE name <> 'World' AND name <> 'Antartica'
 ORDER BY water_land_ratio DESC
 LIMIT 10;

 * sqlite:///factbook.db
Done.


[('British Indian Ocean Territory', 905.6666666666666),
 ('Virgin Islands', 4.520231213872832),
 ('Puerto Rico', 0.5547914317925592),
 ('Bahamas, The', 0.3866133866133866),
 ('Guinea-Bissau', 0.2846728307254623),
 ('Malawi', 0.25939625850340137),
 ('Netherlands', 0.22571032366565366),
 ('Uganda', 0.22292237442922375),
 ('Eritrea', 0.16435643564356436),
 ('Liberia', 0.15623961794019933)]

The highest average water area to land area ratios belong to relatively small islands; Malawi is the country with the highest water to land ratio that isn't an island or group of islands. Only 2 countries/territories contain more water than land.

In [14]:
%%sql

SELECT COUNT(*)
  FROM facts
 WHERE name <> 'World' AND name <> 'Antartica';

 * sqlite:///factbook.db
Done.


[(260,)]

## Countries with the highest population density

In [15]:
%%sql

SELECT name, CAST(population AS FLOAT) / area AS pop_density
  FROM facts
 WHERE name <> 'World' AND name <> 'Antartica'
 ORDER BY pop_density DESC
 LIMIT 10;

 * sqlite:///factbook.db
Done.


[('Macau', 21168.964285714286),
 ('Monaco', 15267.5),
 ('Singapore', 8141.279770444763),
 ('Hong Kong', 6445.041516245487),
 ('Gaza Strip', 5191.819444444444),
 ('Gibraltar', 4876.333333333333),
 ('Bahrain', 1771.8592105263158),
 ('Maldives', 1319.6409395973155),
 ('Malta', 1310.01582278481),
 ('Bermuda', 1299.9259259259259)]

This list is quite different to the list we got previously, and mostly consists of very small countries.

## Countries that have a higher death rate than birth rate

In [16]:
%%sql

SELECT name, death_rate, birth_rate
  FROM facts
 WHERE death_rate > birth_rate;

 * sqlite:///factbook.db
Done.


[('Austria', 9.42, 9.41),
 ('Belarus', 13.36, 10.7),
 ('Bosnia and Herzegovina', 9.75, 8.87),
 ('Bulgaria', 14.44, 8.92),
 ('Croatia', 12.18, 9.45),
 ('Czech Republic', 10.34, 9.63),
 ('Estonia', 12.4, 10.51),
 ('Germany', 11.42, 8.47),
 ('Greece', 11.09, 8.66),
 ('Hungary', 12.73, 9.16),
 ('Italy', 10.19, 8.74),
 ('Japan', 9.51, 7.93),
 ('Latvia', 14.31, 10.0),
 ('Lithuania', 14.27, 10.1),
 ('Moldova', 12.59, 12.0),
 ('Monaco', 9.24, 6.65),
 ('Poland', 10.19, 9.74),
 ('Portugal', 11.02, 9.27),
 ('Romania', 11.9, 9.14),
 ('Russia', 13.69, 11.6),
 ('Serbia', 13.66, 9.08),
 ('Slovenia', 11.37, 8.42),
 ('Ukraine', 14.46, 10.72),
 ('Saint Pierre and Miquelon', 9.72, 7.42)]

In [17]:
%%sql

SELECT name, death_rate, birth_rate, CAST(death_rate AS FLOAT) / birth_rate AS death_birth_ratio
  FROM facts
 WHERE death_rate > birth_rate AND name <> 'World'
ORDER BY death_birth_ratio DESC;

 * sqlite:///factbook.db
Done.


[('Bulgaria', 14.44, 8.92, 1.6188340807174888),
 ('Serbia', 13.66, 9.08, 1.5044052863436124),
 ('Latvia', 14.31, 10.0, 1.431),
 ('Lithuania', 14.27, 10.1, 1.4128712871287128),
 ('Hungary', 12.73, 9.16, 1.3897379912663756),
 ('Monaco', 9.24, 6.65, 1.3894736842105262),
 ('Slovenia', 11.37, 8.42, 1.350356294536817),
 ('Ukraine', 14.46, 10.72, 1.3488805970149254),
 ('Germany', 11.42, 8.47, 1.3482880755608027),
 ('Saint Pierre and Miquelon', 9.72, 7.42, 1.3099730458221026),
 ('Romania', 11.9, 9.14, 1.3019693654266957),
 ('Croatia', 12.18, 9.45, 1.288888888888889),
 ('Greece', 11.09, 8.66, 1.2806004618937643),
 ('Belarus', 13.36, 10.7, 1.2485981308411216),
 ('Japan', 9.51, 7.93, 1.1992433795712485),
 ('Portugal', 11.02, 9.27, 1.1887810140237325),
 ('Russia', 13.69, 11.6, 1.1801724137931033),
 ('Estonia', 12.4, 10.51, 1.1798287345385348),
 ('Italy', 10.19, 8.74, 1.165903890160183),
 ('Bosnia and Herzegovina', 9.75, 8.87, 1.0992108229988726),
 ('Czech Republic', 10.34, 9.63, 1.0737279335410175),
 ('Moldova', 12.59, 12.0, 1.0491666666666666),
 ('Poland', 10.19, 9.74, 1.0462012320328542),
 ('Austria', 9.42, 9.41, 1.0010626992561105)]

Bulgaria has the highest death rate, and the countries that have the highest death to birth rate ratios are mostly based in Europe.

In [18]:
%%sql

SELECT name, death_rate, birth_rate, CAST(death_rate AS FLOAT) / birth_rate AS death_birth_ratio
  FROM facts
 WHERE name <> 'World'
ORDER BY death_birth_ratio;

 * sqlite:///factbook.db
Done.


[('Kosovo', None, None, None),
 ('Holy See (Vatican City)', None, None, None),
 ('Ashmore and Cartier Islands', None, None, None),
 ('Christmas Island', None, None, None),
 ('Cocos (Keeling) Islands', None, None, None),
 ('Coral Sea Islands', None, None, None),
 ('Heard Island and McDonald Islands', None, None, None),
 ('Norfolk Island', None, None, None),
 ('Clipperton Island', None, None, None),
 ('French Southern and Antarctic Lands', None, None, None),
 ('Saint Barthelemy', None, None, None),
 ('Saint Martin', None, None, None),
 ('Niue', None, None, None),
 ('Tokelau', None, None, None),
 ('Bouvet Island', None, None, None),
 ('Jan Mayen', None, None, None),
 ('Svalbard', None, None, None),
 ('Akrotiri', None, None, None),
 ('British Indian Ocean Territory', None, None, None),
 ('Dhekelia', None, None, None),
 ('Pitcairn Islands', None, None, None),
 ('South Georgia and South Sandwich Islands', None, None, None),
 ('Navassa Island', None, None, None),
 ('Wake Island', None, None, None),
 ('United States Pacific Island Wildlife Refuges', None, None, None),
 ('Antarctica', None, None, None),
 ('Paracel Islands', None, None, None),
 ('Spratly Islands', None, None, None),
 ('Arctic Ocean', None, None, None),
 ('Atlantic Ocean', None, None, None),
 ('Indian Ocean', None, None, None),
 ('Pacific Ocean', None, None, None),
 ('Southern Ocean', None, None, None),
 ('Gaza Strip', 3.04, 31.11, 0.0977177756348441),
 ('Kuwait', 2.18, 19.91, 0.10949271722752386),
 ('Iraq', 3.77, 31.45, 0.11987281399046106),
 ('United Arab Emirates', 1.97, 15.43, 0.12767336357744655),
 ('Oman', 3.36, 24.44, 0.13747954173486088),
 ('Jordan', 3.79, 25.37, 0.14938904217579818),
 ('Solomon Islands', 3.85, 25.77, 0.14939852541715173),
 ('West Bank', 3.5, 22.99, 0.15224010439321445),
 ('Qatar', 1.53, 9.84, 0.15548780487804878),
 ('Vanuatu', 4.09, 25.04, 0.16333865814696485),
 ('Marshall Islands', 4.21, 25.6, 0.16445312499999998),
 ('Timor-Leste', 6.1, 34.16, 0.17857142857142858),
 ('Saudi Arabia', 3.33, 18.51, 0.179902755267423),
 ('Syria', 4.0, 22.17, 0.18042399639152007),
 ('Algeria', 4.31, 23.67, 0.18208702999577522),
 ('Guatemala', 4.77, 24.89, 0.1916432302129369),
 ('Turks and Caicos Islands', 3.1, 16.13, 0.1921884686918785),
 ('Bahrain', 2.69, 13.66, 0.19692532942898974),
 ('Libya', 3.58, 18.03, 0.1985579589572934),
 ('Malawi', 8.41, 41.56, 0.20235803657362847),
 ('Northern Mariana Islands', 3.71, 18.32, 0.20251091703056767),
 ('Brunei', 3.52, 17.32, 0.20323325635103925),
 ('Micronesia, Federated States of', 4.23, 20.54, 0.20593962999026294),
 ('American Samoa', 4.75, 22.89, 0.2075141983398864),
 ('Egypt', 4.77, 22.9, 0.20829694323144105),
 ('Madagascar', 6.81, 32.61, 0.20883164673413063),
 ('Yemen', 6.28, 29.98, 0.209472981987992),
 ('Tonga', 4.85, 23.0, 0.21086956521739128),
 ('Sao Tome and Principe', 7.24, 34.23, 0.2115103710195735),
 ('Togo', 7.26, 34.13, 0.21271608555522997),
 ('Ethiopia', 8.19, 37.27, 0.2197477864233968),
 ('Tanzania', 8.0, 36.39, 0.21984061555372356),
 ('Burundi', 9.27, 42.01, 0.2206617472030469),
 ('South Sudan', 8.18, 36.91, 0.22162015713898675),
 ('Honduras', 5.17, 23.14, 0.22342264477095936),
 ('Benin', 8.21, 36.02, 0.22792892837312603),
 ('Gambia, The', 7.15, 30.86, 0.23169151004536617),
 ('Ghana', 7.22, 31.09, 0.23222901254422643),
 ('Nauru', 5.87, 24.95, 0.23527054108216433),
 ('Belize', 5.97, 24.68, 0.2418962722852512),
 ('Dominican Republic', 4.55, 18.73, 0.24292578750667376),
 ('Uganda', 10.69, 43.79, 0.244119662023293),
 ('Senegal', 8.46, 34.52, 0.2450753186558517),
 ('Equatorial Guinea', 8.19, 33.31, 0.2458721104773341),
 ('Maldives', 3.89, 15.75, 0.246984126984127),
 ('Eritrea', 7.52, 30.0, 0.25066666666666665),
 ('Philippines', 6.11, 24.27, 0.2517511330861146),
 ('Tajikistan', 6.18, 24.38, 0.2534864643150123),
 ('Samoa', 5.32, 20.87, 0.2549113560134164),
 ('Malaysia', 5.03, 19.71, 0.2552004058853374),
 ('Kenya', 6.89, 26.4, 0.2609848484848485),
 ('Mauritania', 8.2, 31.34, 0.2616464582003829),
 ('Sudan', 7.66, 29.19,

The country/territory with the lowest death to birth ratio is the Gaza Strip. 

## Countries that will add the most people to their populations

In [19]:
%%sql

SELECT *, (population*(population_growth/100)) AS pop_growth
  FROM facts
 WHERE name <> 'World'
ORDER BY pop_growth DESC
LIMIT 10;

 * sqlite:///factbook.db
Done.


[(77, 'in', 'India', 3287263, 2973193, 314070, 1251695584, 1.22, 19.55, 7.32, 0.04, 15270686.124799998),
 (37, 'ch', 'China', 9596960, 9326410, 270550, 1367485388, 0.45, 12.49, 7.53, 0.44, 6153684.246),
 (129, 'ni', 'Nigeria', 923768, 910768, 13000, 181562056, 2.45, 37.64, 12.9, 0.22, 4448270.372),
 (132, 'pk', 'Pakistan', 796095, 770875, 25220, 199085847, 1.46, 22.58, 6.49, 1.54, 2906653.3662),
 (58, 'et', 'Ethiopia', 1104300, None, 104300, 99465819, 2.89, 37.27, 8.19, 0.22, 2874562.1691),
 (14, 'bg', 'Bangladesh', 148460, 130170, 18290, 168957745, 1.6, 21.14, 5.61, 0.46, 2703323.92),
 (186, 'us', 'United States', 9826675, 9161966, 664709, 321368864, 0.78, 12.49, 8.15, 3.86, 2506677.1392),
 (78, 'id', 'Indonesia', 1904569, 1811569, 93000, 255993674, 0.92, 16.72, 6.37, 1.16, 2355141.8008),
 (40, 'cg', 'Congo, Democratic Republic of the', 2344858, 2267048, 77810, 79375136, 2.45, 34.88, 10.07, 0.27, 1944690.8320000002),
 (138, 'rp', 'Philippines', 300000, 298170, 1830, 100998376, 1.61, 24.27, 6.11, 2.09, 1626073.8536)]

India is due to add the most people to their population, followed by China.

## Future questions?